Source/Core/VideoBackends/Software/CopyRegion.h

```diff
@@ -1,32 +1,29 @@
 #pragma once
 
-#include "Common/MathUtil.h"
-
 #include <cmath>
+#include <vector>
```

Comment:
This include doesn't seem necessary

Reply:
Yep, it's a leftover from testing.  Though it's worth noting that both uses of `CopyRegion` start with vectors and use `data()` for it, so there isn't any strong reason to use raw pointers here (apart from performance maybe, but it's the software renderer...)

Reply:
It was likely to keep it from needing to depend on a single container type, making it more flexible, since it can copy to any contiguous region of memory.

Reply:
Similarly... why does `ScaleTexture` make copies into and out of vectors?  Wouldn't something like

```cpp
  const SWTexture* software_source_texture = static_cast<const SWTexture*>(src_texture);
  SWTexture* software_dest_texture = static_cast<SWTexture*>(dst_framebuffer->GetColorAttachment());

  CopyRegion(software_source_texture->GetData(), src_rect, software_dest_texture->GetData(), dst_rect);
```

work?  (Ignoring issues with the rectangles themselves).

Reply:
Oh, it's because `GetData()` returns a pointer to `u8` while this is supposed to operate on a pointer to `Pixel` (or something else) with a larger size.  It still seems awkward to make a copy; is that *actually* required?

Reply:
degasus, Stenzek, or some other graphic dev would be way more qualified to answer the *why* here than me. But at a glance, I agree.